File: backend/recipe-service/src/main/java/dev/recipeservice/entity/Recipe.java
package dev.recipeservice.entity;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.List;


@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "recipes")
public class Recipe {
    @Id
    private String id;
    private String title;
    private String description;
    private List<String> ingredients;
    private List<String> steps;
    private Integer cookingTimeMinutes;
    private Difficulty difficulty;
    private String imageUrl;
    private Instant createdAt;
    private Instant updatedAt;
}

File: frontend/recipe/src/types/recipe.ts
export type Difficulty = "EASY" | "MEDIUM" | "HARD"

export interface Recipe {
    id: string
    title: string
    description: string
    ingredients: string[]
    steps: string[]
    cookingTimeMinutes: number
    difficulty: Difficulty
    imageUrl?: string
    createdAt?: string
}

export interface RecipeRequest {
    title: string
    description: string
    ingredients: string[]
    steps: string[]
    cookingTimeMinutes: number
    difficulty: Difficulty
}

export interface PageResponse<T> {
    content: T[]
    totalPages: number
    totalElements: number
    size: number
    number: number
}